### PR TITLE
tx.origin misuse example

### PR DESCRIPTION
**Pull request by adamskrodzki**

### CODE REVIEW

**Comment by adamskrodzki** [comment]
looks like something is broken in license/cla 
I've clicked sign and form redirected me by check is still pending

**Comment by disconnect3d** [comment]
@adamskrodzki as the bot says:
> Your Name seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please add the email address used for this commit to your account.

It seems you fired more or less such command on ur machine:
```
git config --global user.name "Your name"
```

Please change your name in git to correspond to your github account:
```
git config --global user.name adamskrodzki
```

And then fix your commits as described here: https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit

**Comment by adamskrodzki** [comment]
@disconnect3d Thank You for your help, looks like now it works

**Comment by dguido** [comment]
Thanks for the PR! Can you include some more detail in your writeup. We want to have the following sections for all of our writeups:

- [ ] Description of the unique vulnerability type
- [ ] Attack scenario(s) to exploit the vulnerability
- [ ] Recommendations to eliminate or mitigate the vulnerability
- [ ] Real-world contracts that exhibit the flaw
- [ ] References to third-party resources with more information

Please also keep the writeup focused on the technical details, and not the state of mind of the developer:

```
Some incompetent developer creates a service FreeSuperSecuredWalletsFactory (Let's call him Danny) 
```

**Comment by adamskrodzki** [comment]
> ```
> Some incompetent developer creates a service FreeSuperSecuredWalletsFactory (Let's call him Danny) 
> ```

Intention was to indroduce real life scenarion. I believe this particular attack is tricky because one 
creating code with it might be neither victim nor attacker and victim might be harmed interacting with some totally different contract. 


Anyway I've edited Readme, please review

File: tx_origin_authorisation_misuse/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# tx.origin misuse
+A Victim contract identifies caller by tx.origin instead of msg.sender 
+That may allow some external malicious smart contract to act on behalf of origin user and attack Victim contract
+
+## Attack Scenario
+
+Some developer (not necesarily attacker, neither victim)  creates a service FreeSuperSecuredWalletsFactory (Let's call him Danny)
```
Comment (disconnect3d) [inline tx_origin_authorisation_misuse/README.md]: Can you try to focus more on technical details? For example it doesn't matter that:
* the service has been created by some developer
* that the developer's name is Danny as he's not mentioned anywhere later on

I.e. the example assumptions should probably be either  rephrased a bit or moved to previous section.

Comment (disconnect3d) [reply tx_origin_authorisation_misuse/README.md]: Also we should format it better, https://guides.github.com/features/mastering-markdown/ along with reviewing other files in this repo might help you with that (there's also a `preview` button to see how your PR would be rendered: [example](https://github.com/trailofbits/not-so-smart-contracts/blob/1913579fe6f4c01fa35b099a9f6898663eea2f71/tx_origin_authorisation_misuse/README.md)).

For example we might introduce bullet points to some of the places and make contract names or functions rendered as inline code like this: `FreeSuperSecuredWalletsFactory`.

**Review by disconnect3d — CHANGES_REQUESTED** [review]

File: tx_origin_authorisation_misuse/README.md (hunk 1 above)
Comment (adamskrodzki) [reply tx_origin_authorisation_misuse/README.md]: take a look if formating is ok now

File: tx_origin_authorisation_misuse/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+## How to reproduce example
+
+Consider using remix.ethereum.org
+
+Deploy FreeSuperSecuredWalletsFactory
+Send Ether to FreeSuperSecuredWalletsFactory from Victim Account
+
+Deploy FakeAirdrop contract passing as parameters
+
+_beneficiary  -  address of atacker
```
Comment (montyly) [inline tx_origin_authorisation_misuse/README.md]: Consider using better markdown formatting here

File: tx_origin_authorisation_misuse/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+_factory - address of  FreeSuperSecuredWalletsFactory
+
+Send transaction to FakeAirdrop  from Victim Account
+
+In result funds sended to FreeSuperSecuredWalletsFactory will be transfered to _beneficiary
```
Comment (montyly) [inline tx_origin_authorisation_misuse/README.md]:
```suggestion
In result funds sended to FreeSuperSecuredWalletsFactory will be transfered to `_beneficiary`
```

File: tx_origin_authorisation_misuse/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+if(listOfWalletsToSteelFrom._wallets(msg.sender)!=address(0)){
+	listOfWalletsToSteelFrom._wallets(msg.sender).sendEth(dishonestGuy);
+}
+```
+
+which drain FreeSuperSecuredWallet of a msg.sender if particular sender owned one. 
+
+## How to reproduce example
+
+Consider using remix.ethereum.org
```
Comment (montyly) [inline tx_origin_authorisation_misuse/README.md]: Consider adding url link in markdown

File: tx_origin_authorisation_misuse/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+In result funds sended to FreeSuperSecuredWalletsFactory will be transfered to _beneficiary
+
+## Real World Example
+
+Mist Wallet in version 0.4.0
+https://blog.ethereum.org/2016/06/24/security-alert-smart-contract-wallets-created-in-frontier-are-vulnerable-to-phishing-attacks/
+
+## Mitigations
+Never use tx.origin as source of user identity
```
Comment (montyly) [inline tx_origin_authorisation_misuse/README.md]:
```suggestion
Never use `tx.origin` as source of user identity
```

File: tx_origin_authorisation_misuse/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+
+## External Resources
+As You can see unsecured wallet can be drained when You interact with completly unrelated service
```
Comment (montyly) [inline tx_origin_authorisation_misuse/README.md]:
```suggestion
As you can see unsecured wallet can be drained when You interact with completly unrelated service
```

**Review by montyly — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution!
Minus the minor typos, it looks good

File: tx_origin_authorisation_misuse/README.md (hunk 2 above)
Comment (adamskrodzki) [reply tx_origin_authorisation_misuse/README.md]: I do not understand what 'markdown formatting' means

**Comment by adamskrodzki** [comment]
Honestly I'm not to much used to use Markdown so it might be misused, now should have been better

**Comment by adamskrodzki** [comment]
@montyly 
@disconnect3d 

Please review I applied Your advices 11 days ago

**Comment by vs77bb** [comment]
Hi @montyly @disconnect3d Vivek from Gitcoin here, are you able to review this one this week? Just checking in on behalf of @adamskrodzki 🙂

**Review by disconnect3d — CHANGES_REQUESTED** [review]
Hey, we have reviewed it once again and we also fixed some markdown issues.

We need better naming and description.

For example, it doesn't look professional when we put emphasis or judge one of the party or the service that takes part in the attack. We don't need any adjectives like `dishonest` or `popular`. It also doesn't matter whether one party spots something or not, the things that matter are actions.

Second of all, please put all code related names in `this form` and not *this form* or **this form**.

Speaking of descriptions - this part is not so easy to understand:
> Now dishonest Drake spot that there is popular service `FreeSuperSecuredWalletsFactory` which has a bug deploys FakeAirdrop contract with his address as _beneficiary, and `FreeSuperSecuredWalletsFactory` address as *_factory* And convince Bob to attend the bounty, and contract indeed do Airdrop of some tokens but at the same time calls following code

Can we try to redact it better (e.g. the sentences are too long)?

**Comment by willsputra** [comment]
@adamskrodzki hey William from Gitcoin here - are you still working on this issue? there are some requested changes from @disconnect3d :)

**Pull request closed without merging**